src/test/integration/pymongo/tests/test_pymongo.py

```diff
@@ -17,23 +17,30 @@ def test_mongo_instrumentation(self):
 
         wait_for_exporter()
 
-        spans_container = SpansContainer.get_spans_from_file()
+        spans = [
+            s
+            for s in SpansContainer.parse_spans_from_file().spans
+            if s["attributes"].get("db.system") == "mongodb"
+        ]
+
+        set_span, get_span = spans[
+            -2:
+        ]  # there are more spans to connect to the mongo container
 
         # assert mongo children spans
-        assert spans_container.find_child_span(
```

Comment:
`spans_container.find_child_span` returned a generator, so this line didn't check anything...